lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -132,20 +129,43 @@ def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
                     "@timestamp": "2021-05-21T24:00:00.141295",
                 },
             }
+
+        Output:
+
+            {
+                "directories":
+                [
+                    "sample1"
+                ],
+                "files":
+                [
+                    {
+                        "name": "result.txt",
+                        "mtime": "2021-09-21T17:46:35",
+                        "size": 0,
+                        "mode": "0o644",
+                        "type": "reg"
+                    },
+                    {
+                        "name": "user-benchmark.cmd",
+                        "mtime": "2021-09-21T17:46:35",
+                        "size": 114,
+                        "mode": "0o755",
+                        "type": "reg"
+                    }
+                ]
+            }
+
         """
         dir_list = []
         file_list = []
         for toc in es_json["hits"]["hits"]:
             directory = toc["_source"]["directory"]
             parent = toc["_source"]["parent"]
             direct = directory.rsplit("/", 1)[0]
-            try:
-                if directory != "/":
-                    dir_list.append(toc["_source"]["name"])
-                if not file_list and (parent == directory or parent == direct):
-                    file_list = toc["_source"]["files"]
-            except KeyError as exc:
-                if exc.args[0] == ["files", "name"]:
-                    pass
+            if directory != "/":
+                dir_list.append(toc["_source"].get("name", "None"))
+            if not file_list and (parent == directory or parent == direct):
+                file_list = toc["_source"].get("files", "None")
```

Comment:
Line 167 probably won't do what you want:  that's going to add a string containing `"None"` to your list of directories...and that result will be hard to differentiate from the case where there actually is a subdirectory named "None".  And, changing the default to an actual `None` won't help, as that will add that value to the list, which will be ugly.  So, I think we want to do something more like this:
```
            if directory != "/" and "name" in toc["_source"]:
                dir_list.append(toc["_source"]["name"]))
```
Peter would probably say that we should use a `try` block here instead of the `in` test 🙂, but I think this is more readable and worth the tiny increase in cost.

Analogously, I don't think that line 169 is going to do what you want:  that's going to set `file_list`, which is supposed to be a `list`, to the string `"None"`.  I think you want the second argument to `get()` to be `[]`, so that, if the `"files"` key is missing, it will assign `file_list` to an empty list.

However, all that said, I suspect that, once we get the query straightened out, this code is going to look a little different:  if the hit is the `parent` directory itself, then we'll capture the file list (and we won't need to test whether we've already set `file_list` because there's only one such hit, and we won't concern ourself with the `dir_list`); otherwise, we'll capture the `"directory"` value (which I'm fairly sure is always there, so we won't need to test for it).  (I'm not sure that we'll need to do anything special for the "root" directory.)